rm102/gui.py

```diff
@@ -50,11 +51,30 @@ def update_gui(self, reg_count, acc, error):
             self.commandRegisterList.model.item(reg_count).setBackground(
                     QBrush(QColorConstants.Red if error else QColorConstants.Red))
 
+    def register_changed(self):
+        self.file_is_open(self.filepath, False)
+
     def runButtonClicked(self):
         self.commandRegisterList.model.run(self.regListModel)
         if self.commandRegisterList.model.stop is not None:
             self.runButton.setEnabled(False)
 
+    def closeEvent(self, event):
+        if not self.changes_saved:
+            filename = filename = self.filepath.split('/')[-1] if self.filepath else 'Unbenannt'
+            msg = QMessageBox()
+            msg.setIcon(QMessageBox.Warning)
+            msg.setWindowTitle("Änderungen speichern?")
+            msg.setText(f"Sollen die Änderungen an {filename} gespeichert werden?")
+            msg.setStandardButtons(QMessageBox.Save | QMessageBox.Discard | QMessageBox.Cancel)
+            msg.setEscapeButton(QMessageBox.Cancel)
+            result = msg.exec()
```

Comment:
I fear many lines here are too long, too :(

Reply:
Solved in 2fef9d64a201e1a54df39048c1c597cd0fe0d8ee